microsoftgraph/microsoft-graph-explorer-v4
# Task: add headers to share query

## Overview

Enables sharing headers with the query being shared.
Fixes #355 

### Demo

![image](https://user-images.githubusercontent.com/58787602/78263699-fa702e00-750a-11ea-8d95-1a5e0d0b2d9f.png)


### Notes

We needed to refactor the request headers component too because the shared headers were not being displayed on the request headers.

Also encountered a bug where query samples with headers within them did not have their headers displayed

We then bundled the headers into the sampleQuery state so that they can be handled easily hence no need for separate action creators and reducers

## Testing Instructions

* Run this query 'Enumerate list items with specific column values'
* Click the share query button and copy the shared url
* Paste the url in a new tab and edit out the production url for the localhost url
* Notice that your headers are displayed in the Request Headers tab
